crypto/store/store_result.c

```diff
@@ -134,6 +134,8 @@ int ossl_store_handle_load_result(const OSSL_PARAM params[], void *arg)
     if (*v == NULL && !try_key(&helper_data, v, ctx, provider, libctx, propq))
         goto err;
     ERR_pop_to_mark();
+    if (*v == NULL && helper_data.object_type == OSSL_OBJECT_PKEY)
```

Comment:
Is this right? Cannot a PKEY be decoded from a PKCS12 file?

Reply:
Unfortunately I cannot recall exactly why I found that change to be needed to prevent spurious error queue entries,
and I am not deep into the details of the loader code.
I agree that PKEYs should be loadable from PKCS12,
but maybe this as been tried sufficiently by the `try_key()` call just before.

Anyway, after rebasing and fixing merge conflicts, I found that in case DSA is disabled, test of 59.30-test_evp.t fails as follows:
```
 ../../util/wrap.pl ../../apps/openssl pkey -in ../../test/certs/server-dsa-pubkey.pem -pubin 2> server-dsa-pubkey.err => 1
Could not read key of Public Key from ../../test/certs/server-dsa-pubkey.pem
not ok 58 - expected error loading unsupported dsa public key
```
and this disappears after reverting the above change.
So I added a respective fixup, hoping that OTOH this does not bring back certain spurious error queue entries.